src/medimail/medimail-client.service.ts

```diff
@@ -0,0 +1,248 @@
+    __dirname,
+    '../wsdl/medimail.wsdl'
+  );
+
+  private _soapClient: Client;
+  private _httpClient: HttpClient;
+  private _wsdl: WSDL;
+  private _acount: string;
+
+  constructor() { }
+
+  public async init(pfx: Buffer, passphrase: string = '', acount = ''): Promise<void> {
+    this._httpClient = new HttpClient();
+    this._acount = acount;
+    this._wsdl = new WSDL(this._wsdlUrl, this._wsdlUrl, {});
+    this._soapClient = await createClientAsync(this._wsdlUrl, {
+      forceSoap12Headers: true,
+      httpClient: this._httpClient,
+    });
+    this._soapClient.setSecurity(
+      new ClientSSLSecurityPFX(pfx, {
+        passphrase,
+      })
+    );
+  }
+
+  /**
+   * Test the api with a simple hello.
+   *
+   * @param name Name of the person to greet
+   * @returns complete once we get the first successful response
+   */
+  public async hello(name: string) {
+    const payload = { name };
+    return this._call(MedimailActions.HELLO, payload);
+  }
+
+  /**
+   * Send a message to the Medimail API.
+   *
+   * @param message The message to send
+   * @returns complete once we get the first successful response
+   */
+  public async send(options: SendMessageOptions): Promise<SendResult> {
+    const { title, message, signatories, recipients,attachments } = options;
+    const joinedSignatories = signatories.join(';');
+    const joinedRecipients = recipients?.join(';');
+    const payload = {
+      message,
+      title,
+      acount: this._acount,
+      signatories: joinedSignatories,
+      recipients: joinedRecipients,
+      attachments,
+    };
+    const reply = await this._call(MedimailActions.SEND, payload);
+    return this._wsdl.xmlToObject(reply.response.formatted.sendResult.$value).webisend;
+  }
+
+  /**
+   * Open a message to the Medimail API.
+   *
+   * @param ref The reference of the message to send
+   * @returns complete once we get the first successful response
+   */
+  public async open(ref: string): Promise<OpenResult> {
+    const payload = {
+      ref,
+      acount: this._acount,
+    };
+    const reply = await this._call(MedimailActions.OPEN, payload);
+    return this._wsdl.xmlToObject(reply.response.formatted.openReturn.$value).webiopen
+  }
+
+  private async _call<T>(
+    action: MedimailActions,
+    soapBody: T
+  ): Promise<FormattedResponse> {
+    return new Promise<FormattedResponse>((resolve, reject) => {
+      this._soapClient[action](
+        soapBody,
+        (
+          err: ISoapError,
+          result: MSSSoapResult,
+          rawResponse: string,
+          // soapHeader: any,
+          rawRequest: string
+        ) => {
+          if (err) {
+            reject(err);
+          }
+          const errorString = MedimailClient.extractErrorIfPresent(result);
+          if (errorString) {
+            reject(new Error(errorString));
+          }
+
+          const formattedResponse: FormattedResponse = {
+            status: 'SUCCESS',
```

Comment:
This structure might be too complex for our case. For INS we needed to preserve all the SOAP requests. We need to check if it is still needed.